metadata-ingestion/src/datahub/sql_parsing/tool_meta_extractor.py

```diff
@@ -42,14 +55,45 @@ class ToolMetaExtractor:
     by warehouse query logs.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, looker_user_mapping: Optional[Dict[str, str]] = None) -> None:
         self.report = ToolMetaExtractorReport()
         self.known_tool_extractors: List[Tuple[str, Callable[[QueryLog], bool]]] = [
             (
                 "mode",
                 self._extract_mode_query,
-            )
+            ),
+            (
+                "looker",
+                self._extract_looker_query,
+            ),
         ]
+        # maps user id (as string) to email address
+        self.looker_user_mapping = looker_user_mapping
+
+    @classmethod
+    def create(
+        cls,
+        config: ToolMetaExtractorConfig,
+        graph: Optional[DataHubGraph] = None,
+    ) -> "ToolMetaExtractor":
+        looker_user_mapping: Dict[str, str] = {}
+        if graph and config.looker_platform_details:
```

Comment:
Is is mandatory to configure `looker_platform_details` ?  What happens, its not set ?

Reply:
Yes we need it for platform-instance

Reply:
I believe we would want the defaults work for no platform instance (no additional user configurations) and work as long as we get single platform resource for looker for platform resource type we need. 

We might need to use `PlatformResource.search_by_key` instead of `PlatformResource.from_datahub` for this to work.

Reply:
Removed